src/app/lib/key.ts

```diff
@@ -1,5 +1,16 @@
 import nacl from 'tweetnacl'
 
+/**
+ * Verify if a key pair is actually usable for signatures
+ */
+function canSignWith(keyPair: nacl.SignKeyPair) {
+  const { secretKey, publicKey } = keyPair
+  const testMessage = Uint8Array.from([1, 2, 3])
+  const signedTestMessage = nacl.sign(testMessage, secretKey)
+  const testDecoded = nacl.sign.open(signedTestMessage, publicKey)
+  return testDecoded && testDecoded.toString() === testMessage.toString()
```

Comment:
:information_source: reason for the `testDecoded &&`

> nacl.sign.open(signedMessage, publicKey)
>
> Verifies the signed message and returns the message without signature.
>
> Returns null if verification failed.

the `.toString()` gets us an easily comparable `1,2,3`

Reply:
SInce the returned value can also be `null`, I can't simply call `.toString()`, but `?.toString()` works, so I got rid of `testDecoded &&`.

Reply:
oops I mean that bit about the toString as a separate note.

could also do `'' + testDecoded` or `String(testDecoded)`

Reply:
So instead of `testDecoded?.toString() === testMessage.toString()`, would you prefer `testDecoded + '' === testMessage + ''` ?

Reply:
really the `testDecoded &&` was fine imo

Reply:
The linter actually came up with this:
```
return `${testDecoded}` === `${testMessage}`
```
This version is probably marginally more efficient (with the string building), and can also handle the null without an extra check.

Reply:
that sounds fine too

Reply:
OK, so let's go with that, and I'm marking this conversation as resolved.